client.go

```diff
@@ -41,7 +42,35 @@ func NewClient(gqlEndpoint string, opt *ClientOpts) *Client {
 	return c
 }
 
+var (
+	ErrHTTPRedirect     = errors.New("redirected request with http status code")
+	ErrHTTPFailedStatus = errors.New("error response with http status code")
```

Comment:
if we are defining errors, i would prefer them to be fully comprehensible on their own:

```suggestion
	ErrHTTPRedirect     = errors.New("http request redirected")
	ErrHTTPRequestFailed = errors.New("http request failed")
```

and used as:

```go
err = fmt.Errorf("%w with status code: %d", ErrHTTPRedirect, resp.StatusCode)
err = fmt.Errorf("%w with status code: %d", ErrHTTPRequestFailed, resp.StatusCode)
```